tests/python/gaia-ui-tests/gaiatest/tests/functional/cost_control/test_cost_control_data_alert_mobile.py

```diff
@@ -46,29 +41,19 @@ def test_cost_control_data_alert_mobile(self):
         search = Search(self.marionette)
         search.launch(launch_timeout=30000)
         browser = search.go_to_url('http://mozqa.com/qa-testcase-data/Images/sample_png_02.png')
-        browser.wait_for_page_to_load(180)
-
-        browser.switch_to_content()
-        Wait(self.marionette, timeout=60).until(lambda m: "sample_png_02.png" in m.title)
-        browser.switch_to_chrome()
+        browser.wait_for_page_to_load(timeout=180)
+        browser.wait_for_title_to_load(title='sample_png_02.png')
 
         system = System(self.marionette)
+        system.wait_for_notification_toaster_displayed()
```

Comment:
Isn't the risk here that the notification toaster has already been displayed while the browser is busy loading (during wait_for_page_load)?